tests/core/channels/test_twilio_voice.py

```diff
@@ -0,0 +1,253 @@
+                "speech_timeout": "auto"
+            }, InvalidConfigException
+        ),
+        (
+            {
+                "initial_prompt": "hello",
+                "reprompt_fallback_phrase": "i didn't get that",
+                "assistant_voice": "woman",
+                "enhanced": "true",
+                "speech_model": "phone_call",
+                "speech_timeout": "auto",
+            }, InvalidConfigException
+        )
+    ]
+)
+def test_invalid_configs(configs: Dict[Text, Any], expected: Type[RasaException]):
+    with pytest.raises(expected):
+        TwilioVoiceInput(**configs)
+
+
+async def test_twilio_voice_remove_image():
+
+    with pytest.warns(UserWarning):
+        output_channel = TwilioVoiceCollectingOutputChannel()
+        await output_channel.send_response(
+            recipient_id="Chuck Norris",
+            message={
+                "image": "https://i.imgur.com/nGF1K8f.jpg",
+                "text": "Some text."
+            }
+        )
+
+
+async def test_twilio_voice_keep_image_text():
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+    await output_channel.send_response(
+        recipient_id="Chuck Norris",
+        message={
+            "image": "https://i.imgur.com/nGF1K8f.jpg",
+            "text": "Some text."
+        }
+    )
+    assert len(output_channel.messages) == 1
+    assert output_channel.messages[0]["text"] == "Some text."
+
+
+async def test_twilio_emoji_warning():
+
+    with pytest.warns(UserWarning):
+        output_channel = TwilioVoiceCollectingOutputChannel()
+        await output_channel.send_response(
+            recipient_id="User",
+            message={
+                "text": "Howdy 😀"
+            }
+        )
+
+
+async def test_twilio_voice_multiple_responses():
+
+    inputs = {
+        "initial_prompt": "hello",
+        "reprompt_fallback_phrase": "i didn't get that",
+        "speech_model": "default",
+        "speech_timeout": "5",
+        "assistant_voice": "woman",
+        "enhanced": "false"
+    }
+
+    tv = TwilioVoiceInput(**inputs)
+
+    output_channel = TwilioVoiceCollectingOutputChannel()
+
+    await output_channel.send_text_message(recipient_id="Chuck Norris", text="message 1")
+    await output_channel.send_text_message(recipient_id="Chuck Norris", text="message 2")
+    assert len(output_channel.messages) == 2
+    assert output_channel.messages[0]["text"] == "message 1"
+    assert output_channel.messages[1]["text"] == "message 2"
+
+    twiml = tv._build_twilio_voice_response(output_channel.messages)
+
+    assert (
+        str(twiml)
+        == '<?xml version="1.0" encoding="UTF-8"?><Response><Say voice="woman">message 1</Say><Pause length="1" /><Gather action="/webhooks/twilio_voice/webhook" actionOnEmptyResult="true" enhanced="false" input="speech" speechModel="default" speechTimeout="5"><Say voice="woman">message 2</Say></Gather></Response>'
+    )
+
+
+async def test_twilio_receive(stack_agent: Agent):
```

Comment:
Can we please add more tests for receive? There is still a whole bunch of uncovered cases (marked by the red lines):
![Screenshot 2021-05-04 at 09 22 00](https://user-images.githubusercontent.com/7667273/116971741-4ff0ae80-acba-11eb-8bee-91efa4b410f4.png)